proxy/http/HttpConfig.h

```diff
@@ -857,6 +857,9 @@ struct HttpConfigParams : public ConfigInfo {
 
   MgmtByte server_session_sharing_pool = TS_SERVER_SESSION_SHARING_POOL_THREAD;
 
+  // Vector to hold the status codes that will BE cached with negative caching enabled
+  std::vector<std::bitset<10>> codeNegCache;
```

Comment:
Why is this a `std::vector`? And why 10 as the base bitset size? Why not 32 or 64? But overall, the maximum possibly status code is 599, which makes `std::bitset<600>` a reasonable choice. Then you can do `c.codeNegCache[HTTP_STATUS_NOT_FOUND] = true;`.

Reply:
P.S. "codeNegCache" is a bad name. "isNegativeStatus" might be better, for something like "if (isNegativeStatus[client.info->status]) { ...}`.

Reply:
I agree. I had some misunderstanding on the vector stuff